pyaedt/edb_core/hfss.py

```diff
@@ -524,6 +525,65 @@ def create_differential_wave_port(
         _edb_boundle_terminal = self._edb.Cell.Terminal.BundleTerminal.Create(edb_list)
         return port_name, ExcitationDifferential(self._pedb, _edb_boundle_terminal)
 
+    @pyaedt_function_handler
+    def create_bundle_wave_port(
+        self,
+        primitives_id: list,
+        points_on_edge: list[list],
+        port_name: str = None,
+        horizontal_extent_factor: str = 5,
+        vertical_extent_factor: str = 3,
+        pec_launch_width: str = "0.01mm",
+    ):
+        """Create a bundle wave port.
+
+        Parameters
+        ----------
+        primitives : list
+            Primitive ID of the positive terminal.
+        points_on_edge : list
+            Coordinate of the point to define the edge terminal.
+            The point must be close to the target edge but not on the two
+            ends of the edge.
+        port_name : str, optional
+            Name of the port. The default is ``None``.
+        horizontal_extent_factor : int, float, optional
+            Horizontal extent factor. The default value is ``5``.
+        vertical_extent_factor : int, float, optional
+            Vertical extent factor. The default value is ``3``.
+        pec_launch_width : str, optional
+            Launch Width of PEC. The default value is ``"0.01mm"``.
+
+        Returns
+        -------
+        tuple
+            The tuple contains: (port_name, pyaedt.edb_core.edb_data.sources.ExcitationDifferential).
+
+        Examples
+        --------
+        >>> edb.core_hfss.create_bundle_wave_port(0, ["-50mm", "-0mm"], 1, ["-50mm", "-0.2mm"])
+        """
+        if not port_name:
+            port_name = generate_unique_name("bundle_port")
+
+        terminals = []
+        _port_name = port_name
+        for p_id, loc in list(zip(primitives_id, points_on_edge)):
+            _, term = self.create_wave_port(
+                p_id,
+                loc,
+                port_name=_port_name,
+                horizontal_extent_factor=horizontal_extent_factor,
+                vertical_extent_factor=vertical_extent_factor,
+                pec_launch_width=pec_launch_width,
+            )
+            _port_name = None
```

Comment:
Do we need to set `_port_name` back to `None` ?

Reply:
yes, the terminals must not have the same name. I set it back to None to let create method to determine unique names

Reply:
It seems in the code that _port_name is not used anymore after this line.

Reply:
It is used in the for loop
_, term = self.create_wave_port(
                p_id,
                loc,
                port_name=**_port_name,**

Reply:
I see it is used to reset the name of the port_name after the first wave port creation.
Make sense @ring630